scipy/stats/_bws_test.py

```diff
@@ -0,0 +1,230 @@
+            Bx_num = (Ri - (m + n) / n * i) * abs(Ri - (m + n) / n * i)
+            By_num = (Hj - (m + n) / m * j) * abs(Hj - (m + n) / m * j)
+            Bx_den = i / (n + 1) * (1 - i / (n + 1)) * m * (m + n) / n
+            By_den = j / (m + 1) * (1 - j / (m + 1)) * n * (m + n) / m
+        else:
+            Bx_num = (Ri - (m + n + 1) / (n + 1) * i) * \
+                abs(Ri - (m + n + 1) / (n + 1) * i)
+            By_num = (Hj - (m + n + 1) / (m + 1) * j) * \
+                abs(Hj - (m + n + 1) / (m + 1) * j)
+            Bx_den = i / (n + 1) * (1 - i / (n + 1)) * \
+                m * (m + n + 1) / (n + 2)
+            By_den = j / (m + 1) * (1 - j / (m + 1)) * \
+                n * (m + n + 1) / (m + 2)
+
+        Bx = 1 / n * np.sum(Bx_num / Bx_den, axis=-1)
+        By = 1 / m * np.sum(By_num / By_den, axis=-1)
+
+        if alternative == 'greater':
+            B = (Bx - By) * 0.5
+        else:
+            B = (By - Bx) * 0.5
+
+    return B
+
+
+BWSResult = namedtuple('BWSResult', ('statistic', 'pvalue'))
+
+
+def bws_test(x, y, axis=0, alternative="two-sided", variant="naive", n_resamples=9999, random_state=None):
+    r'''Perform the Baumgartner-Weiss-Schindler test on two independent samples.
+
+    The Baumgartner-Weiss-Schindler (BWS) test is a nonparametric test of 
+    the null hypothesis that the distribution underlying sample `x` 
+    is the same as the distribution underlying sample `y`. Unlike 
+    the Kolmogorv-Smirnov, Wilcoxon, and Cramer-Von Mises tests, 
+    the BWS test weights the integral by the variance of the difference
+    in CDFs, emphasizing the tails of the distributions, which increases
+    the power of the test for a lot of applications.
+
+    Parameters
+    ----------
+    x, y : array-like
+        N-d arrays of samples. The arrays must be broadcastable except along
+        the dimension given by `axis`.
+    axis : int, optional
+        Axis along which to perform the test. Default is 0.
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis. Default is 'two-sided'.
+        Let *F(u)* and *G(u)* be the cumulative distribution functions of the
+        distributions underlying `x` and `y`, respectively. Then the following
+        alternative hypotheses are available:
+
+        * 'two-sided': the distributions are not equal, i.e. *F(u) ≠ G(u)* for
+          at least one *u*.
+        * 'less': the distribution underlying `x` is stochastically less
+          than the distribution underlying `y`, i.e. *F(u) >= G(u)* for all *u*.
+        * 'greater': the distribution underlying `x` is stochastically greater
+          than the distribution underlying `y`, i.e. *F(u) <= G(u)* for all *u*.
+
+        Under a more restrictive set of assumptions, the alternative hypotheses
+        can be expressed in terms of the locations of the distributions;
+        see [3] section 5.1.
+    variant : {'shan', 'murakami', 'naive'}, optional
```

Comment:
Something I asked in the original PR - is having these variants worth the extra complexity?
Only one person has asked for this test in the past few decades, so I wonder whether the variant would be missed. 

If the "naive" option is naive and missing something that murakami accounts for, why not only use that?

What are your thoughts?

Reply:
yeah, it might be better